### ISSUE TITLE
"dev.containers.executeInWSL": false doesn't work at all

### ISSUE THREAD

**Issue by TerrorJack** [microsoft/vscode-remote-release#9194]
- VSCode Version: 1.84.0
- Local OS Version: Windows 11 23H2 (OS Build 22631.2506)
- Remote OS Version: Ubuntu 22.04
- Remote Extension/Connection Type: Containers

- Logs:

```
[33 ms] Dev Containers 0.321.0 in VS Code 1.84.0 (d037ac076cee195194f93ce6fe2bdfe2969cc82d).
[32 ms] Start: Resolving Remote
[34 ms] Setting up container: /terrorjack
[34 ms] Context: build01
[42 ms] Start: Run: C:\ProgramData\chocolatey\bin\docker.exe inspect --type container /terrorjack
[1839 ms] Start: Run: wsl -l -v
[1939 ms] Start: Run: wsl -d NixOS -e /bin/sh -c echo ~
[2096 ms] Start: Run: wsl -d NixOS -e /bin/sh -c cd '/home/terrorjack' && /bin/sh
[2104 ms] Start: Run in host: id -un
[2223 ms] terrorjack
[2224 ms] 
[2224 ms] Start: Run in host:  (command -v getent >/dev/null 2>&1 && getent passwd 'terrorjack' || grep -E '^terrorjack|^[^:]*:[^:]*:terrorjack:' /etc/passwd || true)
[2228 ms] Start: Run in host: echo ~
[2229 ms] /home/terrorjack
[2230 ms] 
[2230 ms] Start: Run in host: test -x '/home/terrorjack/.vscode-remote-containers/bin/d037ac076cee195194f93ce6fe2bdfe2969cc82d/node'
[2231 ms] 
[2232 ms] 
[2232 ms] Exit code 1
[2232 ms] Start: Run in host: test -x '/home/terrorjack/.vscode-server/bin/d037ac076cee195194f93ce6fe2bdfe2969cc82d/node'
[2234 ms] 
[2234 ms] 
[2234 ms] Start: Run in host: test -f '/home/terrorjack/.vscode-server/bin/d037ac076cee195194f93ce6fe2bdfe2969cc82d/node_modules/node-pty/package.json'
[2236 ms] 
[2236 ms] 
[2236 ms] Start: Run in host: test -f '/home/terrorjack/.vscode-remote-containers/dist/vscode-remote-containers-server-0.321.0.js'
[2238 ms] 
[2238 ms] 
[2241 ms] userEnvProbe: loginInteractiveShell (default)
[2241 ms] userEnvProbe: not found in cache
[2242 ms] userEnvProbe shell: /run/current-system/sw/bin/bash
[2327 ms] userEnvProbe PATHs:
Probe:     '/run/wrappers/bin:/nix/store/vwkvhj69z4qqgmpa2lwm97kabf12p26r-coreutils-9.3/bin:/nix/store/qyzfglbrqb5ck0dgljplin2bvc4995w7-findutils-4.9.0/bin:/nix/store/p2r51wfg9m3ga7pp7avslpfhfa7w5y83-gnugrep-3.11/bin:/nix/store/f5qy259g9b4qh0hwz22z5j5bq3m53cpv-gnutar-1.35/bin:/nix/store/kmr52zpw7wazxywqvzgpdx0vnn9prd3v-gzip-1.13/bin:/nix/store/vapnrxrw3b21c7ji61bmnbzl2cj1vl96-systemd-254.3/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/wsl/lib:/home/terrorjack/.nix-profile/bin:/nix/profile/bin:/home/terrorjack/.local/state/nix/profile/bin:/etc/profiles/per-user/terrorjack/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin'
Container: None
[2328 ms] Start: Check Docker is running
[2329 ms] Start: Run: C:\ProgramData\chocolatey\bin\docker.exe version --format {{.Server.APIVersion}}
[2949 ms] Server API version: 1.41
[2950 ms] Start: Run: C:\Program Files\Microsoft VS Code\Code.exe --ms-enable-electron-run-as-node c:\Users\terrorjack\.vscode\extensions\ms-vscode-remote.remote-containers-0.321.0\dist\spec-node\devContainersSpecCLI.js set-up --container-id c5dd41da47826d51b2b80f5e66a44c28437e2afc6e625fcd41d7be07696d05b8 --user-data-folder c:\Users\terrorjack\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-containers\data --docker-path C:\ProgramData\chocolatey\bin\docker.exe --container-session-data-folder /tmp/devcontainers-0445bcff-6858-428b-afd7-972dbb316cb61699556676377 --log-level debug --log-format json --config c:\Users\terrorjack\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-containers\imageConfigs\terrorjack.json --default-user-env-probe loginInteractiveShell --skip-post-create --terminal-columns 195 --terminal-rows 22
[3199 ms] @devcontainers/cli 0.52.1. Node.js v18.15.0. win32 10.0.22631 x64.
[3199 ms] Start: Run: C:\ProgramData\chocolatey\bin\docker.exe buildx version
[3286 ms] 
[3286 ms] docker: 'buildx' is not a docker command.
See 'docker --help'

[3286 ms] Exit code 1
[3288 ms] Start: Run: C:\ProgramData\chocolatey\bin\docker.exe inspect --type container c5dd41da47826d51b2b80f5e66a44c28437e2afc6e625fcd41d7be07696d05b8
[4652 ms] Start: Inspecting container
[4652 ms] Start: Run: C:\ProgramData\chocolatey\bin\docker.exe inspect --type container c5dd41da47826d51b2b80f5e66a44c28437e2afc6e625fcd41d7be07696d05b8
[5268 ms] Start: Run in container: /bin/sh
[5274 ms] Start: Run in container: uname -m
[7421 ms] x86_64
[7421 ms] 
[7421 ms] Start: Run in container: (cat /etc/os-release || cat /usr/lib/os-release) 2>/dev/null
[7455 ms] PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
[7455 ms] 
[7455 ms] Start: Run in container:  (command -v getent >/dev/null 2>&1 && getent passwd 'terrorjack' || grep -E '^terrorjack|^[^:]*:[^:]*:terrorjack:' /etc/passwd || true)
[7489 ms] Start: Run in container: test -f '/var/devcontainer/.patchEtcEnvironmentMarker'
[7519 ms] 
[7519 ms] 
[7519 ms] Start: Run in container: test -f '/var/devcontainer/.patchEtcProfileMarker'
[7549 ms] 
[7549 ms] 
[7563 ms] Start: Run: C:\Program Files\Microsoft VS Code\Code.exe --ms-enable-electron-run-as-node c:\Users\terrorjack\.vscode\extensions\ms-vscode-remote.remote-containers-0.321.0\dist\spec-node\devContainersSpecCLI.js read-configuration --docker-path C:\ProgramData\chocolatey\bin\docker.exe --container-id c5dd41da47826d51b2b80f5e66a44c28437e2afc6e625fcd41d7be07696d05b8 --log-level debug --log-format json --config c:\Users\terrorjack\AppData\Roaming\Code\User\globalStorage\ms-vscode-remote.remote-containers\imageConfigs\terrorjack.json --include-merged-configuration --mount-workspace-git-root --terminal-columns 195 --terminal-rows 22
[7817 ms] @devcontainers/cli 0.52.1. Node.js v18.15.0. win32 10.0.22631 x64.
[7817 ms] Start: Run: C:\ProgramData\chocolatey\bin\docker.exe inspect --type container c5dd41da47826d51b2b80f5e66a44c28437e2afc6e625fcd41d7be07696d05b8
[9202 ms] Start: Inspecting container
[9202 ms] Start: Run: C:\ProgramData\chocolatey\bin\docker.exe inspect --type container c5dd41da47826d51b2b80f5e66a44c28437e2afc6e625fcd41d7be07696d05b8
[10630 ms] Start: Run in container: /bin/sh
[10636 ms] Start: Run in container: uname -m
[12848 ms] x86_64
[12848 ms] 
[12849 ms] Start: Run in container: (cat /etc/os-release || cat /usr/lib/os-release) 2>/dev/null
[13378 ms] PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
[13379 ms] 
[13379 ms] Start: Run in container:  (command -v getent >/dev/null 2>&1 && getent passwd 'terrorjack' || grep -E '^terrorjack|^[^:]*:[^:]*:terrorjack:' /etc/passwd || true)
[13619 ms] Start: Run in container: test -d /home/terrorjack/.vscode-server
[13724 ms] 
[13724 ms] 
[13725 ms] Start: Run in container: test ! -f '/home/terrorjack/.vscode-server/data/Machine/.writeMachineSettingsMarker' && set -o noclobber && mkdir -p '/home/terrorjack/.vscode-server/data/Machine' && { > '/home/terrorjack/.vscode-server/data/Machine/.writeMachineSettingsMarker' ; } 2> /dev/null
[13834 ms] 
[13835 ms] 
[13835 ms] Exit code 1
[13835 ms] Start: Run in container: cat /home/terrorjack/.vscode-server/data/Machine/settings.json
[13945 ms] 
[13945 ms] cat: /home/terrorjack/.vscode-server/data/Machine/settings.json: No such file or directory
[13945 ms] Exit code 1
[13945 ms] Start: Run in container: test -d /home/terrorjack/.vscode-server/bin/d037ac076cee195194f93ce6fe2bdfe2969cc82d
[14052 ms] 
[14052 ms] 
[14052 ms] Start: Launching Dev Containers helper.
[14053 ms] ssh-agent: SSH_AUTH_SOCK not set on local host.
[14053 ms] ssh-agent: SSH_AUTH_SOCK in container (/tmp/vscode-ssh-auth-0f85b362-327b-42d5-85ce-17ebd254385f.sock) forwarded to local host (\\.\pipe\openssh-ssh-agent).
[14053 ms] X11 forwarding: DISPLAY not set on local host.
[14053 ms] Start: Run in container: test -e /tmp/.X11-unix/X0
[14154 ms] 
[14154 ms] 
[14155 ms] Start: Run in container: test -e /tmp/.X11-unix/X1
[14251 ms] 
[14251 ms] 
[14251 ms] Start: Run in container: test -e /tmp/.X11-unix/X2
[14343 ms] 
[14343 ms] 
[14343 ms] Start: Run in container: test -e /tmp/.X11-unix/X3
[14436 ms] 
[14436 ms] 
[14436 ms] Start: Run in container: test -e /tmp/.X11-unix/X4
[15013 ms] 
[15014 ms] 
[15014 ms] Start: Run in container: test -e /tmp/.X11-unix/X5
[15123 ms] 
[15123 ms] 
[15123 ms] Start: Run in container: test -e /tmp/.X11-unix/X6
[15229 ms] 
[15230 ms] 
[15230 ms] Start: Run in container: test -e /tmp/.X11-unix/X7
[15323 ms] 
[15323 ms] 
[15323 ms] Start: Run in container: test -e /tmp/.X11-unix/X8
[15422 ms] 
[15422 ms] 
[15423 ms] Start: Run in container: test -e /tmp/.X11-unix/X9
[15533 ms] 
[15533 ms] 
[15534 ms] Start: Run in container: test -e /tmp/.X11-unix/X10
[15635 ms] 
[15635 ms] 
[15636 ms] Start: Run in container: test -e /tmp/.X11-unix/X11
[15736 ms] 
[15737 ms] 
[15737 ms] Start: Run in container: test -e /tmp/.X11-unix/X12
[15835 ms] 
[15836 ms] 
[15836 ms] Start: Run in container: test -e /tmp/.X11-unix/X13
[15947 ms] 
[15948 ms] 
[15948 ms] Start: Run in container: test -e /tmp/.X11-unix/X14
[16511 ms] 
[16512 ms] 
[16512 ms] Start: Run in container: test -e /tmp/.X11-unix/X15
[16616 ms] 
[16616 ms] 
[16616 ms] Start: Run in container: test -e /tmp/.X11-unix/X16
[16721 ms] 
[16721 ms] 
[16721 ms] Start: Run in container: test -e /tmp/.X11-unix/X17
[16835 ms] 
[16836 ms] 
[16837 ms] Start: Run in container: test -e /tmp/.X11-unix/X18
[16944 ms] 
[16944 ms] 
[16944 ms] Start: Run in container: test -e /tmp/.X11-unix/X19
[17049 ms] 
[17049 ms] 
[17049 ms] Start: Run in container: test -e /tmp/.X11-unix/X20
[17148 ms] 
[17149 ms] 
[17149 ms] Start: Run in container: test -e /tmp/.X11-unix/X21
[17241 ms] 
[17242 ms] 
[17242 ms] Start: Run in container: test -e /tmp/.X11-unix/X22
[17478 ms] 
[17479 ms] 
[17479 ms] Start: Run in container: test -e /tmp/.X11-unix/X23
[17581 ms] 
[17582 ms] 
[17582 ms] Start: Run in container: test -e /tmp/.X11-unix/X24
[17678 ms] 
[17678 ms] 
[17678 ms] Start: Run in container: test -e /tmp/.X11-unix/X25
[17780 ms] 
[17780 ms] 
[17780 ms] Start: Run in container: test -e /tmp/.X11-unix/X26
[17886 ms] 
[17887 ms] 
[17887 ms] Exit code 1
[17887 ms] Start: Run in container: mkdir -p '/tmp/.X11-unix'
[18001 ms] 
[18002 ms] 
[18002 ms] X11 forwarding: DISPLAY in container (:26) forwarded to wsl host (:0).
[18002 ms] Start: Run in container: gpgconf --list-dir agent-socket
[18116 ms] /home/terrorjack/.gnupg/S.gpg-agent
[18116 ms] 
[18116 ms] Start: Run in container: gpgconf --list-dir homedir
[18218 ms] /home/terrorjack/.gnupg
[18218 ms] 
[18218 ms] Start: Run in container: ls '/home/terrorjack/.gnupg/private-keys-v1.d' 2>/dev/null
[18321 ms] 
[18321 ms] 
[18321 ms] Exit code 2
[18322 ms] Start: Run: gpgconf --list-dir agent-extra-socket
[18338 ms] gpg-agent: No agent-extra-socket found on local host.
[18339 ms] Start: Run in Host: gpgconf --list-dir agent-extra-socket
[18344 ms] Host server: Error: spawn gpgconf ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:284:19)
    at onErrorNT (node:internal/child_process:477:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
[18346 ms] gpg-agent: No agent-extra-socket found on wsl host.
[18347 ms] Start: Run in container: /bin/sh
[18347 ms] userEnvProbe: loginInteractiveShell (default)
[18347 ms] Start: Run in container: test -f '/tmp/devcontainers-0445bcff-6858-428b-afd7-972dbb316cb61699556676377/env-loginInteractiveShell.json'
[18351 ms] Start: Run in container: echo ~
[18352 ms] Host server: (node:35464) PromiseRejectionHandledWarning: Promise rejection was handled asynchronously (rejection id: 1)
(Use `node --trace-warnings ...` to show where the warning was created)
[18457 ms] 
[18457 ms] 
[18457 ms] Exit code 1
[18457 ms] Start: Run in container: # Test for /home/terrorjack/.ssh/known_hosts and ssh
[18457 ms] userEnvProbe: not found in cache
[18457 ms] userEnvProbe shell: /bin/bash
[18559 ms] /home/terrorjack/.ssh/known_hosts exists
[18560 ms] 
[18560 ms] Exit code 1
[18560 ms] Start: Run in container: for pid in `cd /proc && ls -d [0-9]*`; do { echo $pid ; readlink /proc/$pid/cwd || echo ; readlink /proc/$pid/ns/mnt || echo ; cat /proc/$pid/stat | tr "
[18856 ms] Start: Run in container: cat '/home/terrorjack/.vscode-server/bin/d037ac076cee195194f93ce6fe2bdfe2969cc82d/product.json'
[19672 ms] Start: Run in container: cat '/home/terrorjack/.vscode-server/data/Machine/.connection-token-d037ac076cee195194f93ce6fe2bdfe2969cc82d' 2>/dev/null || (umask 377 && echo '1bbc533f-62c7-4c92-ba8f-37b0c8e62f57' >'/home/terrorjack/.vscode-server/data/Machine/.connection-token-d037ac076cee195194f93ce6fe2bdfe2969cc82d-74895937-5e0e-40a3-bc41-5831b1612276' && mv -n '/home/terrorjack/.vscode-server/data/Machine/.connection-token-d037ac076cee195194f93ce6fe2bdfe2969cc82d-74895937-5e0e-40a3-bc41-5831b1612276' '/home/terrorjack/.vscode-server/data/Machine/.connection-token-d037ac076cee195194f93ce6fe2bdfe2969cc82d' && rm -f '/home/terrorjack/.vscode-server/data/Machine/.connection-token-d037ac076cee195194f93ce6fe2bdfe2969cc82d-74895937-5e0e-40a3-bc41-5831b1612276' && cat '/home/terrorjack/.vscode-server/data/Machine/.connection-token-d037ac076cee195194f93ce6fe2bdfe2969cc82d')
[19709 ms] /home/terrorjack
[19709 ms] 
[19710 ms] Start: Run in container: cat <<'EOF-/tmp/vscode-remote-containers-0f85b362-327b-42d5-85ce-17ebd254385f.js' >/tmp/vscode-remote-containers-0f85b362-327b-42d5-85ce-17ebd254385f.js
[19742 ms] 
[19743 ms] 
[19744 ms] Start: Run in container: cat <<'EOF-/tmp/vscode-remote-containers-server-0f85b362-327b-42d5-85ce-17ebd254385f.js' >/tmp/vscode-remote-containers-server-0f85b362-327b-42d5-85ce-17ebd254385f.js_1699556696584
[19779 ms] 5f67d778-e64b-4b4c-89c3-3743b4cb23a1
[19779 ms] 
[19780 ms] Start: Starting VS Code Server
[19781 ms] Start: Preparing Extensions
[19781 ms] Start: Run in container: test ! -f '/home/terrorjack/.vscode-server/data/Machine/.installExtensionsMarker' && set -o noclobber && mkdir -p '/home/terrorjack/.vscode-server/data/Machine' && { > '/home/terrorjack/.vscode-server/data/Machine/.installExtensionsMarker' ; } 2> /dev/null
[19836 ms] 
[19837 ms] 
[19888 ms] 
[19888 ms] 
[19889 ms] Exit code 1
[19890 ms] Extensions cache, install extensions: None
[19890 ms] Start: Run in container: test -d /home/terrorjack/.vscode-server/extensionsCache && ls /home/terrorjack/.vscode-server/extensionsCache || true
[20000 ms] userEnvProbe PATHs:
Probe:     '/home/terrorjack/.local/bin:/home/terrorjack/.cargo/bin:/home/terrorjack/.ghcup/bin:/home/terrorjack/.deno/bin:/home/terrorjack/.elan/bin:/home/terrorjack/.opam/default/bin:/home/terrorjack/.pack/bin:/home/terrorjack/.vcpkg:/home/terrorjack/.local/bin:/nix/var/nix/profiles/per-user/terrorjack/profile/bin:/usr/lib/llvm-17/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin'
Container: '/home/terrorjack/.cargo/bin:/home/terrorjack/.ghcup/bin:/home/terrorjack/.deno/bin:/home/terrorjack/.elan/bin:/home/terrorjack/.opam/default/bin:/home/terrorjack/.pack/bin:/home/terrorjack/.vcpkg:/home/terrorjack/.local/bin:/nix/var/nix/profiles/per-user/terrorjack/profile/bin:/usr/lib/llvm-17/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin'
Merged:    '/home/terrorjack/.local/bin:/home/terrorjack/.cargo/bin:/home/terrorjack/.ghcup/bin:/home/terrorjack/.deno/bin:/home/terrorjack/.elan/bin:/home/terrorjack/.opam/default/bin:/home/terrorjack/.pack/bin:/home/terrorjack/.vcpkg:/home/terrorjack/.local/bin:/nix/var/nix/profiles/per-user/terrorjack/profile/bin:/usr/lib/llvm-17/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin'
[20002 ms] arrterian.nix-env-selector-1.0.10
asciidoctor.asciidoctor-vscode-3.1.7
denoland.vscode-deno-3.26.0
dhall.dhall-lang-0.0.4
eamodio.gitlens-14.4.1
haskell.haskell-2.4.1
justusadam.language-haskell-3.6.0
llvm-vs-code-extensions.vscode-clangd-0.1.24
ms-python.black-formatter-2023.6.0
ms-python.python-2023.20.0
ms-python.vscode-pylance-2023.11.10
ms-toolsai.jupyter-2023.10.1003070148-linux-x64
ms-toolsai.vscode-jupyter-cell-tags-0.1.8
ms-vscode.wordcount-0.1.0
ocamllabs.ocaml-platform-1.13.4
rust-lang.rust-analyzer-0.3.1722-linux-x64
stkb.rewrap-1.16.3
timonwong.shellcheck-0.34.0-linux-x64
trond-snekvik.simple-rst-1.5.4
vadimcn.vscode-lldb-1.10.0
[20002 ms] 
[20002 ms] Start: Run in container: mkdir -p '/tmp/devcontainers-0445bcff-6858-428b-afd7-972dbb316cb61699556676377' && cat > '/tmp/devcontainers-0445bcff-6858-428b-afd7-972dbb316cb61699556676377/env-loginInteractiveShell.json' << 'envJSON'
[20004 ms] Extensions cache, copy to remote: None
[20005 ms] Optimizing extensions for quality: stable
[20006 ms] Start: Run in container: /home/terrorjack/.vscode-server/bin/d037ac076cee195194f93ce6fe2bdfe2969cc82d/bin/code-server --log debug --force-disable-user-env --server-data-dir /home/terrorjack/.vscode-server --telemetry-level all --accept-server-license-terms --host 127.0.0.1 --port 0 --connection-token-file /home/terrorjack/.vscode-server/data/Machine/.connection-token-d037ac076cee195194f93ce6fe2bdfe2969cc82d --extensions-download-dir /home/terrorjack/.vscode-server/extensionsCache --start-server 
[20099 ms] 
[20100 ms] 
[21353 ms] *
* Visual Studio Code Server
*
* By using the software, you agree to
* the Visual Studio Code Server License Terms (https://aka.ms/vscode-server-license) and
* the Microsoft Privacy Statement (https://privacy.microsoft.com/en-US/privacystatement).
*
[21358 ms] Server bound to 127.0.0.1:36513 (IPv4)
Extension host agent listening on 36513

[21358 ms] Start: Run in container: echo 36513 >'/home/terrorjack/.vscode-server/data/Machine/.devport-d037ac076cee195194f93ce6fe2bdfe2969cc82d'
[21463 ms] 
[21463 ms] 
[21464 ms] Port forwarding for container port 36513 starts listening on local port.
[21466 ms] Port forwarding local port 36513 to container port 36513
[21466 ms] Git credential helper not enabled.
[21467 ms] Start: Run in container: /bin/sh -c # Watch installed extensions
[21468 ms] Start: Run in container: /bin/sh -c # Watch machine settings
[21479 ms] Port forwarding connection from 65133 > 36513 > 36513 in the container.
[21480 ms] Start: Run in container: /home/terrorjack/.vscode-server/bin/d037ac076cee195194f93ce6fe2bdfe2969cc82d/node -e 
[21497 ms] [19:04:58] 




[21523 ms] [19:04:58] Extension host agent started.
[21577 ms] [19:04:58] Removing uninstalled extensions: [ 'vadimcn.vscode-lldb-1.9.2' ]
[24341 ms] Port forwarding 65133 > 36513 > 36513 stderr: Connection established
[24364 ms] Port forwarding connection from 65144 > 36513 > 36513 in the container.
[24364 ms] Start: Run in container: /home/terrorjack/.vscode-server/bin/d037ac076cee195194f93ce6fe2bdfe2969cc82d/node -e 
[24500 ms] [19:05:01] [127.0.0.1][fc35613f][ManagementConnection] New connection established.
[24712 ms] [19:05:01] Log level changed to info
[26470 ms] Port forwarding 65144 > 36513 > 36513 stderr: Connection established
[26535 ms] Port forwarding connection from 65151 > 36513 > 36513 in the container.
[26536 ms] Start: Run in container: /home/terrorjack/.vscode-server/bin/d037ac076cee195194f93ce6fe2bdfe2969cc82d/node -e 
[26668 ms] Port forwarding connection from 65156 > 36513 > 36513 in the container.
[26672 ms] Start: Run in container: /home/terrorjack/.vscode-server/bin/d037ac076cee195194f93ce6fe2bdfe2969cc82d/node -e 
[27034 ms] [19:05:04] [127.0.0.1][242a6fd6][ExtensionHostConnection] New connection established.
[27054 ms] [19:05:04] [127.0.0.1][242a6fd6][ExtensionHostConnection] <3333879> Launched Extension Host Process.
[27838 ms] Port forwarding 65151 > 36513 > 36513 stderr: Connection established
[29953 ms] Port forwarding 65156 > 36513 > 36513 stderr: Connection established
[30102 ms] Extensions cache, remote removals: None
[32860 ms] Port forwarding 65151 > 36513 > 36513 stderr: Remote close
[32917 ms] Port forwarding 65151 > 36513 > 36513 terminated with code 0 and signal null.
[34973 ms] Port forwarding 65156 > 36513 > 36513 stderr: Remote close
[35116 ms] Port forwarding 65156 > 36513 > 36513 terminated with code 0 and signal null.
[68490 ms] Port forwarding 65151 > 36513 > 36513: Local close
[68490 ms] Port forwarding 65156 > 36513 > 36513: Local close
[83795 ms] Start: Run in container: /bin/sh -c # Watch machine settings
[84342 ms] Start: Run in container: /bin/sh -c # Watch installed extensions
[90220 ms] Extensions cache, remote removals: None
[146528 ms] Start: Run in container: /bin/sh -c # Watch machine settings
[146763 ms] Start: Run in container: /bin/sh -c # Watch installed extensions
[150327 ms] Extensions cache, remote removals: None
[208394 ms] Start: Run in container: /bin/sh -c # Watch machine settings
[209388 ms] Start: Run in container: /bin/sh -c # Watch installed extensions
[210443 ms] Extensions cache, remote removals: None
```

Steps to Reproduce:

1. Install only the Docker CLI tools on Windows, not the full Docker desktop environment
2. Set up Docker context to make it connect to a remote Docker daemon based on instructions at https://code.visualstudio.com/docs/containers/ssh
3. Attach to a running container

Both Docker CLI and OpenSSH are win32 native programs, so it makes no sense that the devcontainers extension would attempt to invoke WSL at all. And I believe `"dev.containers.executeInWSL"` settings exists for this purpose, but even when it's explicitly set to `false`, the log still shows attempts to launch WSL.

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

**Comment by joecarl** [comment]
Similar problem here when trying to start a Devcontainer while connected to another server via Remote-SSH extension. It always worked fine for me, but last week it started to fail. Checking logs I realized it was due to VS Code trying to start WSL. Why? I don't even have WSL installed on my computer!

**Comment by bamurtaugh** [comment]
Thanks folks for filing and commenting.

Do you still experience this with the latest Dev Containers extension and VS Code Insiders?

**Comment by chrmarti** [comment]
`"dev.containers.executeInWSL"` controls where `docker` is run. We still connect to WSL for display, ssh-agent and gpg-agent forwarding.

**Comment by yuuuxt** [comment]
Having the same issue. Never changed `dev.containers.executeInWSL` and checked it's kept unticked.  

Using win10, vscode 1.85.0. the last working version is `0.309.0`, and the latest pre-release `0.328.0` doesn't work.

**Comment by chrmarti** [comment]
@yuuuxt Could you open a new issue with the log from when this happens? (`F1` > `Dev Containers: Show Container Log`) It sounds like this might be a different issue.

**Comment by yuuuxt** [comment]
I assume it's the same issue, as the solution is exactly the same - downgrading to `0.309.0`.  more details in #9324

**Comment by joecarl** [comment]
> Thanks folks for filing and commenting.
> 
> Do you still experience this with the latest Dev Containers extension and VS Code Insiders?

Yup. A few months later and the error persists, even with VS Code Insiders.
- OS: `Windows 11`
- VS Code Version: `1.86.0-insider`
- Dev Containers Extension Version: `0.335.0 (pre-release)`
- Log:
```
[68 ms] Dev Containers 0.335.0 over Remote - SSH 0.107.1 in VS Code 1.86.0-insider (1091f68d834974d78cd4f9769034b4ae4581ff2a).
[68 ms] Start: Resolving Remote
[79 ms] Setting up container for folder or workspace: /home/user/repos/test
[79 ms] Host authority: ssh-remote+user@192.168.10.132
[89 ms] Start: Run: wsl -l -v
[165 ms] Could not connect to WSL.
[165 ms] Command failed: wsl -l -v
```
I insist this happens while being connected to a remote linux server via remote-ssh. The error must be because it is trying to start WSL🧐 (look at the logs). 
Since my remote server is already a linux machine with docker installed it should directly run docker, not WSL!
(I have no wsl installed anywhere)

As some of us stated, the latest working version is `0.309.0`.

Does anybody know any workaround (apart from downgrading)?

Thanks for your awesome work!

**Comment by chrmarti** [comment]
I'm adding a setting to turn off forwarding of WSL services. Please give that a try in the next Dev Containers pre-release (likely 0.337.0-pre-release).

**Comment by chrmarti** [comment]
Dev Containers 0.337.0-pre-release (currently building, should be available within half a hour) is adding a user setting to turn forwarding of WSL services off:
![Image](https://github.com/microsoft/vscode-remote-release/assets/9205389/11d242c1-9657-4370-a433-84012f19f368)

Please give that a try. Note that this version currently requires VS Code Insiders (it won't be available for installation with VS Code 1.85): http://code.visualstudio.com/insiders/
